tests/array.rs

```diff
@@ -716,7 +716,7 @@ fn diag() {
     let a = arr2(&[[1., 2., 3.0f32], [0., 0., 0.]]);
     let d = a.view().into_diag();
     assert_eq!(d.dim(), 2);
-    let d = arr2::<f32, _>(&[[]]).into_diag();
+    let d = arr2::<f32, 0>(&[[]]).into_diag();
```

Comment:
The zero can't be inferred here? That's worrying but if it's just particular to zero maybe it's no big deal.

Reply:
Const generic arguments can't be inferred with _ in general.

Reply:
Sad times